flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/test/program/TableApiTestStep.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.test.program;
+
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.api.TableResult;
+import org.apache.flink.table.types.AbstractDataType;
+
+import java.util.function.Function;
+
+/** Test step for execution of a Table API. Similar to {@link SqlTestStep}. */
+public class TableApiTestStep implements TestStep {
+    private final Function<TableEnvAccessor, Table> tableQuery;
+    private final String sinkName;
+
+    TableApiTestStep(Function<TableEnvAccessor, Table> tableQuery, String sinkName) {
```

Comment:
We recently introduced `TablePipeline`, maybe this is a better parameter and avoids having the additional `sinkName` parameter:

```
(Function<TableEnvAccessor, TablePipeline> tablePipeline)
```

Reply:
Unfortunately `TablePipeline` does not give me access to the underlying `QueryOperation`. I need that in tests so that I can extract it and convert to SQL string by calling `asSerializableString`.

I'd need to cast `TablePipeline` to `TablePipelineImpl`. If that's ok with you I can refactor to that.